0000-constrained-backends.md

```diff
@@ -0,0 +1,110 @@
+- Feature Name: constrained backends
+- Start Date: 2023-06-23
+- SEP Status: Draft
+- SEP PR: (leave this empty)
+- Salt Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Apply constraints to individual backends to allow collaboration with "foreign" backends.
+
+# Motivation
+[motivation]: #motivation
+
+Why are we doing this? What use cases does it support? What is the expected outcome?
+
+This will open parts of the configuration for other developers (maybe in another department or even external) to modify, but without giving them root rights, even if the minion runs as root.
+Currently the publisher_acl is only applied via the salt api - but if a state enters the salt master via a (git) backend, there is no constraint checking, allowing anyone that can change an backend to basically change almost everything, if not everything.
+For example: This hinders to allow "foreign" git repositories to be used as backend without extensive merge approval process.
+
+In the end, an application administrator could contribute states / functions / ... that are specific to a software and enable more complex orchestrations, without having a merge approver to know everything about every application to spot (unintentional or intentional) malicious changes and an expert keeps being an expert for just one/assigned scope/s, without the need to be trusted with everything.
+
+# Design
+[design]: #detailed-design
+
+This is the bulk of the SEP. Explain the design in enough detail for somebody familiar
+with the product to understand, and for somebody familiar with the internals to implement. It should include:
+
+- Definition of any new terminology
+  - "foreign" backend: a backend that can be modified by other users than the salt master administrator.
+- Examples of how the feature is used:
+      - Target minions on backends
+        Users able to change webserver.git aren't always the same users that should be able to modify backupservers.
+      - Apply these states constrained to be applied as user x.
```

Comment:
How? Fork a minion process running as the target user to run it? Most stuff would break, such as communication with the master.

Reply:
"Target minions on backends" would modify (enable / disable) overlays of the fileserver on the base of targeting - which is controlled by the master. It would allow to create a "view" on a environment. This is maybe a feature on its own.

The latter "Apply these states constrained to be applied as user x" is tricky . as it would have to postprocess the output of a renderer, to ensure something like "runas" is set.
As for communication: The minion would have to act as proxy/filter for the unprivileged subprocess.